### PR TITLE
Add numpy array support for bqplot linking

### PR DESCRIPTION
**Pull request by martinRenou**

### CODE REVIEW

File: ipysheet/easy.py [hunk 1]
```diff
@@ -208,6 +208,15 @@ def cell_range(value, row_start=0, column_start=0, row_end=None, column_end=None
         A CellRange widget.
     """
     global _cells
+
+    try:
+        import numpy as np
```
Comment (martinRenou) [inline ipysheet/easy.py]: I think the `try except` should go away. It's not a big deal if ipysheet has NumPy for dependency, right? We can add it

**Comment by maartenbreddels** [comment]
What do you think of this implementation, it should work with anything that has a to_numpy or tolist method, which means ndarrays, series, dataframes.

My doubt is on if this should go into the validator or the in the serializer, i,e:
```python
cell_vector.value = x
assert cell_vector.value == x.tolist()  # do we want this? 
assert cell_vector.value is x  # or this?
```

The mock kernel is there to force the path to go into a call of serializating into a json format.

**Comment by martinRenou** [comment]
I don't know, maybe the second option (implementation in the serializer) is better? At least the user would not be surprised because the cell value doesn't change just after it was assigned.

File: ipysheet/sheet.py [hunk 2]
```diff
@@ -35,6 +36,16 @@ class Cell(widgets.Widget):
...
+
+        original_value = value
+
+        # a pandas dataframe will hit this path first
+        if hasattr(value, "to_numpy"):
+            value = value.to_numpy()
+        # numpy arrays or pandas series will hit this path
+        if hasattr(value, "tolist"):
+            value = value.tolist()
+
```
Comment (martinRenou) [inline ipysheet/sheet.py]: @maartenbreddels we want to remove that no? It will be done later in the serializer

File: ipysheet/serializer.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        # numpy arrays or pandas series will hit this path
+        if hasattr(value, "tolist"):
+            value = value.tolist()
+        return value
+
+    def json_to_value(data, widget):
+        # first put pack widgets in
+        value = widgets.widget_serialization['from_json'](data, widget)
+        original = getattr(widget, name)
+        if hasattr(original, 'copy'):  # this path will try to preserve the type
```
Comment (martinRenou) [inline ipysheet/serializer.py]: This is a weak test don't you think? Maybe we should check the value type?

File: ipysheet/sheet.py (hunk 2 above)
Comment (martinRenou) [reply ipysheet/sheet.py]: No we need it actually, for the shape checking

File: ipysheet/serializer.py (hunk 3 above)
Comment (martinRenou) [reply ipysheet/serializer.py]: I think it is fine for now, I'll open an issue just to be sure not to forget it

**Pull request merged**